src/main/java/org/wildfly/security/auth/util/ElytronAuthenticator.java

```diff
@@ -104,7 +104,18 @@ protected PasswordAuthentication getPasswordAuthentication() {
                 }
             }
         } catch (IOException | NoSuchAlgorithmException | InvalidKeySpecException e) {
-            return null;
+            if (e instanceof InvalidKeySpecException) {
```

Comment:
Creating the callback with ```ClearPassword.ALGORITHM_CLEAR``` doesn't seem to work. What we can do is change ```ElytronXmlParser``` to actually create a ```ClearPassword``` instead of a ```RawPassword```.

Or am I missing something ?